### ISSUE TITLE
Folder Contents: List folder contents and keep in sync with core data.

### ISSUE THREAD

**Issue by aerych** [Automattic/newspack-ios#58]
Display the contents of a folder in a list. 
There should be a matching core data representation backing the contents of a folder.  This will require some updates to the current folder code.

Folders and folder contents should:
- Detect when a new folder or asset is added and create any needed data.
- Detect when an existing folder or asset is deleted and clean up any associated data.
- Detect when an existing folder is moved/renamed and maintain the correct data association.

Basically what's in core data should be kept in sync with what's in the file system.

Additionally: 
- Associate Folders and contents with a site.  Site folders naming should be robust.
